airflow/api_fastapi/common/parameters.py

```diff
@@ -406,6 +407,37 @@ def depends(self, dag_id: str | None = None) -> _DagIdFilter:
         return self.set_value(dag_id)
 
 
+class _UriPatternSearch(_SearchParam):
+    """Search on uri."""
+
+    def __init__(self, skip_none: bool = True) -> None:
+        super().__init__(AssetModel.uri, skip_none)
+
+    def depends(self, uri_pattern: str | None = None) -> _UriPatternSearch:
+        return self.set_value(uri_pattern)
+
+
+class _DagIdAssetReferencePatternSearch(_SearchParam):
```

Comment:
I don't think this should be called `Search` or Extend `_SearchParam`. It does not use anything from the Search, (no text matching, ilike etc..).

It's more `Filter` to me.

Reply:
Yeah that makes sense. Making the change and calling it `_DagIdAssetReferenceFilter` while making it extedn from `BaseParam`